### PR TITLE
firewall: T4178: Use lowercase for TCP flags and add an validator

### PR DESCRIPTION
**Pull request by sarthurdev**
## Change Summary

* Fix for case sensitivity in nftables config
* Add tcp flags validator

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4178

## Component(s) name

firewall, policy

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
I wonder if we can not use the new firewall systems migration script whichvisbin place to also migrate the flags from upper to lowercase in the CLI?

File: interface-definitions/include/firewall/common-rule.xml.i [hunk 1]
```diff
@@ -280,6 +280,9 @@
           <format> </format>
           <description>\n\n  Allowed values for TCP flags : SYN ACK FIN RST URG PSH ALL\n  When specifying more than one flag, flags should be comma-separated.\n For example : value of 'SYN,!ACK,!FIN,!RST' will only match packets with\n  the SYN flag set, and the ACK, FIN and RST flags unset</description>
         </valueHelp>
```
Comment (c-po) [inline interface-definitions/include/firewall/common-rule.xml.i:278]: Please also add valueHelp for each individual flag and also a completionHelper list so tab complete will work :)

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by sarthurdev** [comment]
Don't think migration is necessary for text case, I think we should handle both and just output to lowercase for nftables.

**Comment by c-po** [comment]
It might work but would be bad practice if CLI accepts both upper and loser case versions ;) - prone to error

**Comment by c-po** [comment]
this needs to be refactored in a later sped, as multiple flags on a comma separated line just feels wrong

**Pull request merged**